pandas/core/frame.py

```diff
@@ -11372,7 +11375,20 @@ def _get_data() -> DataFrame:
                     row_index = np.tile(np.arange(nrows), ncols)
                     col_index = np.repeat(np.arange(ncols), nrows)
                     ser = Series(arr, index=col_index, copy=False)
-                    result = ser.groupby(row_index).agg(name, **kwds)
+                    # GroupBy will raise a warning with SeriesGroupBy as the object,
+                    # likely confusing users
+                    with rewrite_warning(
```

Comment:
is this context expensive?  if so might make sense to only do it in idxmin/idxmax instead of in _reduce? (though this is only for axis=1 so NBD?)

Reply:
It seems to be very cheap:

```
size = 1000000
df = pd.DataFrame({'a': np.random.randint(0, 100, size), 'b': np.random.randint(0, 100, size)})
df['a'] = df['a'].mask(df['a'] < 50, np.nan)
df['b'] = df['b'].mask(df['b'] < 50, np.nan)
%timeit df.idxmin(axis=1)

# Without context
# 111 ms ± 669 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)
# With context
# 113 ms ± 887 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)
```

In any case, if we were to try to move this any lower I think we'd need to detect who the caller is? We only want to replace this warning when groupby is called from _reduce.

Reply:
Ah - I see, use e.g. a nullcontext for other ops. I think this is cheap enough but can implement if you'd prefer.

Reply:
what i had in mind was not having any context here but instead in `(Series|DataFrame).idx(min|max)`.  But if you say its cheap then its fine